PennMobile/New Group/PAC Code/PacCodeViewController.swift

```diff
@@ -0,0 +1,150 @@
+    
+    var pacCode : String?
+        
+    lazy var quadDigitLabel = [digitLabel, digitLabel, digitLabel, digitLabel]
+    
+    let pacCodeHStack = UIStackView()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        view.backgroundColor = .white
+        
+        self.title = "PAC Code"
+        
+        self.navigationItem.rightBarButtonItem = .init(barButtonSystemItem: .refresh, target: self, action: #selector(refreshButtonTapped))
+        
+        setupPacCodeHStack()
+        setupPacCodeInfoLabel()
+    }
+    
+    var digitLabel : UILabel {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.heightAnchor.constraint(equalToConstant: 70).isActive = true
+        label.widthAnchor.constraint(equalToConstant: 70).isActive = true
+        
+        label.textAlignment = .center
+        label.font = UIFont.boldSystemFont(ofSize: 30)
+        
+        label.backgroundColor = .grey6
+        
+        return label
+    }
+
+    func setupPacCodeHStack() {
+        pacCodeHStack.axis = .horizontal
+        pacCodeHStack.distribution = .equalCentering
+        pacCodeHStack.spacing = 10.0
+        
+        for digitLabel in quadDigitLabel {
+            pacCodeHStack.addArrangedSubview(digitLabel)
+        }
+        
+        view.addSubview(pacCodeHStack)
+        pacCodeHStack.translatesAutoresizingMaskIntoConstraints = false
+        pacCodeHStack.centerYAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerYAnchor).isActive = true
+        pacCodeHStack.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor).isActive = true
+    }
+    
+    func setupPacCodeInfoLabel() {
+        let pacCodeInfoLabel = UILabel()
+        pacCodeInfoLabel.textColor = .labelPrimary
+        pacCodeInfoLabel.font = UIFont.systemFont(ofSize: 35)
+        pacCodeInfoLabel.text = "Your PAC Code is"
+        
+        view.addSubview(pacCodeInfoLabel)
+        pacCodeInfoLabel.translatesAutoresizingMaskIntoConstraints = false
+        pacCodeInfoLabel.bottomAnchor.constraint(equalTo: pacCodeHStack.topAnchor, constant: -30).isActive = true
+        pacCodeInfoLabel.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor).isActive = true
+    }
+    
+    func refreshPacCode() {
+        if let pacCode = pacCode {
+            var counter = 0
+            for pacCodeDigit in pacCode {
+                quadDigitLabel[counter].text = "\(pacCodeDigit)"
+                counter += 1
+            }
+        } else {
+            for box in quadDigitLabel {
+                box.text = ""
+            }
+        }
+    }
+}
+
+// MARK: - Local Authentication
+extension PacCodeViewController : LocalAuthentication {
+        
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        handleAuthentication(cancelText: "Go Back", reasonText: "Authenticate to see your PAC Code")
```

Comment:
This function is a bit of misnomer. You're not handling authentication, you're requesting authentication. Let's change this to reflect that.